syntax = "proto3";

package bookstore;

service BookCatalog {
  rpc GetBookInfo(BookRequest) returns (BookResponse);
  rpc ListBooks(Empty) returns (BookList);
}

message BookRequest {
  string title = 1;
}

message BookResponse {
  string title = 1;
  string author = 2;
  int32 year = 3;
  int32 stock = 4;
  float price = 5;
}

message Book {
  string title = 1;
  string author = 2;
  int32 year = 3;
  int32 stock = 4;
  float price = 5;
}

message BookList {
  repeated Book books = 1;
}

message Empty {}
